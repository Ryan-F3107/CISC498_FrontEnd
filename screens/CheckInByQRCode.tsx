import React from 'react';
import { Text, View, TouchableOpacity, Alert } from 'react-native';
import { TextInput, IconButton } from 'react-native-paper';
import styles from '../styling/styles';

class CheckInByQRCode extends React.Component {
    constructor(props) {
        super(props);
        const initalState = {
            numPeople: '',
            num: 0,
            errorNumPeople: ''

        }
        this.state = initalState;
    }

    numUp() {
        this.setState({ num: this.state.num + 1 })
    }

    numDown() {
        if (this.state.num != 0) {
            this.setState({ num: this.state.num - 1 })
        }
    }

    render() {
        return (
            <View style={styles.homeContainer}>
                <IconButton
                    style={styles.closeButton}
                    icon="close"
                    size={35}
                    color={'black'}
                    onPress={() => { this.props.navigation.replace("Home") }}
                ></IconButton>

                <View style={styles.checkInContainer}>
                    <Text style={styles.checkInTitle}> Check-in </Text>
                    <TextInput
                        style={styles.signUpTextInput}
                        label="Number of People In Party"
                        mode="outlined"
                        theme={{ colors: { primary: 'blue' } }}
                        placeholder="0"
                        keyboardType="number-pad"
                        onChangeText={(numPeople) => this.setState({ numPeople: numPeople.replace(/[a-zA-z!@#$%^&*()_=+;.,><?/'|-]/gi, '') })}
                        value={this.state.numPeople}
                        onBlur={() => {
                            if (this.state.numPeople == "") {
                                this.setState(() => ({ errorNumPeople: "Required" }))
                            }
                        }}
                        onFocus={() => {
                            this.setState(() => ({ errorNumPeople: "" }))
                        }}
                    />
                    <Text style={{ color: 'red' }}>{this.state.errorNumPeople}</Text>
                    <View style={{ flexDirection: 'row', alignSelf: 'center' }}>
                        <TouchableOpacity
                            style={styles.NumIncButton}
                            onPress={() => this.numDown()} >
                            <Text style={{ color: "white", alignSelf: "center" }}>-</Text>
                        </TouchableOpacity>
                        <Text style={{ marginTop: 10 }}>{this.state.num}</Text>
                        <TouchableOpacity
                            style={styles.NumIncButton}
                            onPress={() => this.numUp()} >
                            <Text style={{ color: "white", alignSelf: "center" }}>+</Text>
                        </TouchableOpacity>



                    </View>

                    <View style={{ marginTop: 100 }}>
                        <TouchableOpacity
                            style={styles.ViewQRCodebutton}
                            onPress={() => this.props.navigation.replace("Home")}
                            disabled={this.state.numPeople == ""}
                        /*{
                            Alert.alert("Checked In!", "",
                                [{
                                    text: "Dismiss"
                                }
                                ]);
                        }*/
                        //disabled={this.checkForm()}
                        >
                            <Text style={{ color: '#fafafa', alignSelf: 'center' }}> Scan QR Code</Text>
                        </TouchableOpacity>
                    </View>
                </View>

            </View >
        )
    }
}

export default CheckInByQRCode;