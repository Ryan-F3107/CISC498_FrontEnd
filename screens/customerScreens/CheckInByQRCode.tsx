import React from 'react';
import { Text, View, TouchableOpacity, Alert } from 'react-native';
import { IconButton } from 'react-native-paper';
import RNPickerSelect from 'react-native-picker-select';
import styles from '../../styling/styles';
import signUpDefaultstyleForPicker from '../../styling/signUpDefaultPicker';
import { showMessage } from 'react-native-flash-message';

class CheckInByQRCode extends React.Component {
    constructor(props) {
        super(props);
        const initalState = {
            numPeople: '',
            num: 0,
            errorNumPeople: ''

        }
        this.state = initalState;
    }

    render() {
        return (
            <View style={styles.homeContainer}>

                {/*Close Button*/}
                <IconButton
                    style={styles.closeButton}
                    icon="close"
                    size={35}
                    color={'black'}
                    onPress={() => {
                        this.props.navigation.goBack()
                    }}
                ></IconButton>


                <View style={styles.checkInContainer}>
                    <Text style={styles.checkInTitle}> Check-in </Text>

                    {/*Select the number of people who came with the customer. 
                      At most 6 people, including the customer is allowed*/}
                    <Text style={{ marginTop: 5, marginBottom: -5, color: '#04074d' }}>NUMBER OF PEOPLE</Text>
                    <Text style={{ marginTop: 5, color: '#04074d', fontSize: 11 }}>including yourself</Text>
                    <View style={styles.viewAndroidOnly}>
                        <RNPickerSelect
                            onValueChange={(numPeople) => this.setState({ numPeople: numPeople })}
                            placeholder={{ label: "Select the number", value: '' }}
                            useNativeAndroidPickerStyle={false}
                            items={[
                                { label: "1", value: '1' },
                                { label: "2", value: '2' },
                                { label: "3", value: '3' },
                                { label: "4", value: '4' },
                                { label: "5", value: '5' },
                                { label: "6", value: '6' }]}
                            style={signUpDefaultstyleForPicker}
                            onClose={() => { // If the field is left blank, show an error message 
                                if (this.state.numPeople == "") {
                                    this.setState(() => ({ errorNumPeople: "Required" }));
                                } else {
                                    this.setState(() => ({ errorNumPeople: "" }));
                                }
                            }}
                            onOpen={() => { // When the field is tapped, remove the error message
                                this.setState(() => ({ errorNumPeople: "" }));
                            }}

                        />
                    </View>
                    <Text style={styles.errorMessage}>{this.state.errorNumPeople}</Text>

                    <View style={{ marginTop: 100 }}>
                        {/* Scan QR Code button*/}
                        <TouchableOpacity
                            style={styles.ViewQRCodebutton}
                            onPress={() => {
                                if (this.state.numPeople == "") {
                                    showMessage({
                                        message: `Error: Incomplete/Invalid. ${'\n'}${'\n'}Please select the number of people in the party.`,
                                        type: "danger",
                                        autoHide: true,
                                        duration: 2500,
                                        backgroundColor: "#ff504a",
                                        color: "#fafafa",
                                        icon: "danger"
                                    });
                                } else {
                                    this.props.navigation.navigate("CameraComponent", { numVisit: this.state.numPeople, receivedUserInfo: this.props.route.params.receivedUserInfo });
                                }
                            }}
                        >
                            <Text style={{ color: '#fafafa', alignSelf: 'center' }}> Scan QR Code</Text>
                        </TouchableOpacity>
                    </View>
                </View>

            </View >
        )
    }
}

export default CheckInByQRCode;