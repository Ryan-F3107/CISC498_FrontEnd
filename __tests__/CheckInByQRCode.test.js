import React from 'react';
import renderer from 'react-test-renderer';
import CheckInByQRCode from '../screens/customerScreens/CheckInByQRCode';

describe("<CheckInByQRCode />", () => {
    it("has 2 children", () => {
        const tree = renderer.create(<CheckInByQRCode />).toJSON();
        expect(tree.children.length).toBe(2);
    });

    it('renders correctly', () => {
        renderer.create(<CheckInByQRCode />);
    });

    it('HelpMyQRCode testing with snapshot', () => {
        const tree = renderer.create(<CheckInByQRCode />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});