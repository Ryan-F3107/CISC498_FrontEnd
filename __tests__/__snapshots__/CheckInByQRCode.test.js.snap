// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckInByQRCode /> HelpMyQRCode testing with snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fafafa",
      "flex": 1,
      "paddingBottom": 20,
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "disabled": undefined,
      }
    }
    accessible={true}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 6,
        "left": 6,
        "right": 6,
        "top": 6,
      }
    }
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "margin": 6,
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 26.25,
            "height": 52.5,
            "width": 52.5,
          },
          undefined,
          Object {
            "alignSelf": "flex-end",
            "marginRight": 10,
            "marginTop": 40,
          },
        ],
      ]
    }
  >
    <View>
      <Text />
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#fafafa",
        "flex": 1,
        "paddingHorizontal": 50,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 30,
          "paddingBottom": 20,
          "paddingTop": 50,
        }
      }
    >
       Check-in 
    </Text>
    <Text
      style={
        Object {
          "color": "#04074d",
          "marginBottom": -5,
          "marginTop": 5,
        }
      }
    >
      NUMBER OF PEOPLE
    </Text>
    <Text
      style={
        Object {
          "color": "#04074d",
          "fontSize": 11,
          "marginTop": 5,
        }
      }
    >
      including yourself
    </Text>
    <View
      style={
        Object {
          "borderColor": "transparent",
          "borderRadius": 0,
          "borderWidth": 0,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignSelf": "stretch",
            },
            undefined,
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="ios_touchable_wrapper"
        >
          <View
            pointerEvents="box-only"
          >
            <TextInput
              allowFontScaling={true}
              editable={false}
              rejectResponderTermination={true}
              style={
                Array [
                  Object {
                    "backgroundColor": "#fafafa",
                    "borderColor": "grey",
                    "borderRadius": 5,
                    "borderWidth": 1,
                    "fontSize": 16,
                    "marginTop": 10,
                    "paddingBottom": 18,
                    "paddingLeft": 10,
                    "paddingTop": 18,
                  },
                  Object {
                    "color": "#c7c7cd",
                  },
                ]
              }
              testID="text_input"
              underlineColorAndroid="transparent"
              value="Select the number"
            />
          </View>
        </View>
        <Modal
          hardwareAccelerated={false}
          onOrientationChange={[Function]}
          supportedOrientations={
            Array [
              "portrait",
              "landscape",
            ]
          }
          testID="ios_modal"
          transparent={true}
          visible={false}
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "opacity": 1,
              }
            }
            testID="ios_modal_top"
          />
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#f8f8f8",
                  "borderTopColor": "#dedede",
                  "borderTopWidth": 1,
                  "flexDirection": "row",
                  "height": 45,
                  "justifyContent": "space-between",
                  "paddingHorizontal": 10,
                  "zIndex": 2,
                },
                undefined,
              ]
            }
            testID="input_accessory_view"
          >
            <View
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderColor": "#a1a1a1",
                        "borderRightWidth": 1.5,
                        "borderTopWidth": 1.5,
                        "height": 15,
                        "width": 15,
                      },
                      undefined,
                      Object {
                        "marginLeft": 11,
                        "transform": Array [
                          Object {
                            "translateY": 4,
                          },
                          Object {
                            "rotate": "-45deg",
                          },
                        ],
                      },
                      undefined,
                      Object {},
                    ]
                  }
                />
              </View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderColor": "#a1a1a1",
                        "borderRightWidth": 1.5,
                        "borderTopWidth": 1.5,
                        "height": 15,
                        "width": 15,
                      },
                      undefined,
                      Object {
                        "marginLeft": 22,
                        "transform": Array [
                          Object {
                            "translateY": -5,
                          },
                          Object {
                            "rotate": "135deg",
                          },
                        ],
                      },
                      undefined,
                      Object {},
                    ]
                  }
                />
              </View>
            </View>
            <View
              accessible={true}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 4,
                  "left": 4,
                  "right": 4,
                  "top": 4,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="done_button"
            >
              <View
                testID="needed_for_touchable"
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#007aff",
                        "fontSize": 17,
                        "fontWeight": "600",
                        "paddingRight": 11,
                        "paddingTop": 1,
                      },
                      undefined,
                      Object {},
                    ]
                  }
                  testID="done_text"
                >
                  Done
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#d0d4da",
                  "justifyContent": "center",
                },
                Object {
                  "height": 215,
                },
                undefined,
              ]
            }
          >
            <View>
              <RNCPicker
                items={
                  Array [
                    Object {
                      "label": "Select the number",
                      "textColor": undefined,
                      "value": "",
                    },
                    Object {
                      "label": "1",
                      "textColor": undefined,
                      "value": "1",
                    },
                    Object {
                      "label": "2",
                      "textColor": undefined,
                      "value": "2",
                    },
                    Object {
                      "label": "3",
                      "textColor": undefined,
                      "value": "3",
                    },
                    Object {
                      "label": "4",
                      "textColor": undefined,
                      "value": "4",
                    },
                    Object {
                      "label": "5",
                      "textColor": undefined,
                      "value": "5",
                    },
                    Object {
                      "label": "6",
                      "textColor": undefined,
                      "value": "6",
                    },
                  ]
                }
                onChange={[Function]}
                selectedIndex={0}
                style={
                  Array [
                    Object {
                      "height": 216,
                    },
                    undefined,
                  ]
                }
                testID="ios_picker"
              />
            </View>
          </View>
        </Modal>
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "red",
          "fontSize": 11,
          "paddingBottom": 0,
        }
      }
    >
      
    </Text>
    <View
      style={
        Object {
          "marginTop": 100,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#0a0540",
            "borderRadius": 25,
            "marginBottom": 40,
            "opacity": 1,
            "padding": 15,
            "width": 150,
          }
        }
      >
        <Text
          style={
            Object {
              "alignSelf": "center",
              "color": "#fafafa",
            }
          }
        >
           Scan QR Code
        </Text>
      </View>
    </View>
  </View>
</View>
`;
